crud-routing/burger.router.js

```diff
@@ -0,0 +1,81 @@
+const burgerRouter = require('express').Router();
+const bodyParser = require('body-parser');
+const fs = require('fs');
+require('dotenv').config();
+const menuFile = process.env.MENU_FILE;
+
+burgerRouter.use(bodyParser.json());
+
+// get all burgers
+burgerRouter.get('/', (req, res) => {
+    const menuFile = process.env.MENU_FILE;
+    const data = JSON.parse(fs.readFileSync(menuFile));
+    res.json(data.burger);
+});
+
+// get burger by id
+burgerRouter.get('/:id', (req, res) => {
+    const menuFile = process.env.MENU_FILE;
+    const data = JSON.parse(fs.readFileSync(menuFile, 'utf8'));
+    const burgerid = parseInt(req.params.id);
+    const burger = data.burger.find(item => item.id === burgerid);
+
+    if (burger) {
+        res.json(burger);
+    } else {
+        res.status(404).json({ error: 'Burger not found' });
+    }
+});
+
+// POST a new burger
+burgerRouter.post('/', (req, res) => {
+    const menuFile = process.env.MENU_FILE;
+    const data = JSON.parse(fs.readFileSync(menuFile, 'utf8'));
+    const { id, name, price } = req.body;
+    let newBurger;
+    if (id) {
+        newBurger = { id, name, price };
+    } else {
+        const newBurgerId = data.counter[0].burger++;
+        newBurger = { id: newBurgerId, name, price }
+    }
+    data.burger.push(newBurger);
+    fs.writeFileSync(menuFile, JSON.stringify(data));
+    res.json(newBurger);
+});
+
+// PUT (update) a burger by ID
+burgerRouter.put('/:id', (req, res) => {
+    const menuFile = process.env.MENU_FILE;
+    const data = JSON.parse(fs.readFileSync(menuFile, 'utf8'));
```

Comment:
If menu file is a json then JSON.parse is not required please google how to import json file in node js application

Reply:
remove JSON.parse